testutil/network/network.go

```diff
@@ -218,6 +218,7 @@ func New(l Logger, baseDir string, cfg Config) (*Network, error) {
 	// only one caller/test can create and use a network at a time
 	l.Log("acquiring test network lock")
 	lock.Lock()
+	defer lock.Unlock()
```

Comment:
This might deviate from the original purpose of this function. What it aims to do is to unlock through `Cleanup` by the caller. It's a global lock. 
I also encountered this error when I ran `semgrep` locally. I am considering whether we should delete the  `testutil/network` directory since we don't use it now. or add the `lock.Unlock` in this function to eliminate the error, just like you did.

Reply:
I have added the `testutil` folder to `.semgrepignore`.